Wangscape/OptionsManager.cpp

```diff
@@ -0,0 +1,54 @@
+#include "OptionsManager.h"
+
+#include <fstream>
+
+#include <boost/filesystem.hpp>
+
+#include <spotify/json.hpp>
+
+#include "codecs/OptionsCodec.h"
+
+OptionsManager::OptionsManager(std::string filename)
+{
+    std::ifstream ifs(filename);
+    if (!ifs.good())
+    {
+        throw std::runtime_error("Could not open options file");
+    }
+
+    std::string str{std::istreambuf_iterator<char>(ifs),
+                    std::istreambuf_iterator<char>()};
+
+    mOptions = spotify::json::decode<Options>(str.c_str());
+    mOptions.filename = filename;
+
+    // TODO(hryniuk): move it elsewhere
+    auto outputDirectory = mOptions.outputDirectory;
```

Comment:
No! They should be in a separate function for now :).

I'm sure that we should move an output directory creation to a different component in the end, but it's not a thing to do now. It's really nasty that during input JSON reading user gets an error about an output directory, even before Wangscape starts looking for any image.